lib/util/LazySet.js

```diff
@@ -187,12 +187,20 @@ class LazySet {
 		return "LazySet";
 	}
 
+	/**
+	 * @param {import("../serialization/ObjectMiddleware").ObjectSerializerContext} context context
+	 */
 	serialize({ write }) {
 		if (this._needMerge) this._merge();
 		write(this._set.size);
 		for (const item of this._set) write(item);
 	}
 
+	/**
+	 * @template T
+	 * @param {import("../serialization/ObjectMiddleware").ObjectDeserializerContext} context context
```

Comment:
@jakebailey We have a bug/problem here if you rewrite this to (at top):
```
/** @typedef {import("../serialization/ObjectMiddleware").ObjectSerializerContext} ObjectSerializerContext */
/** @typedef {import("../serialization/ObjectMiddleware").ObjectDeserializerContext} ObjectSerializerContext */
```

`LazySet` starts acting weird (one from many erros):
```
lib/sharing/resolveMatchedConfigs.js:88:42 - error TS2345: Argument of type 'LazySet<string>' is not assignable to parameter of type 'Iterable<string> | LazySet<string>'.
```

Reply:
I was able to minimize it, but, it's not new to TS 5.0 it seems.

Reply:
@jakebailey Can you provide a link to the issue?

Reply:
Just filed: https://github.com/microsoft/TypeScript/issues/53967